core/modules/qdisp/QueryResource.h

```diff
@@ -72,10 +72,12 @@ class QueryResource : public XrdSsiService::Resource {
     friend class QueryResourceDebug;
 
 private:
+    char const* _getRname(std::string const& rName);
     XrdSsiSession* _xrdSsiSession {nullptr}; ///< unowned, do not delete.
     std::shared_ptr<JobQuery> _jobQuery;
     std::string const _jobIdStr; ///< for debugging only
     util::InstanceCount _instC{"QueryResource"};
+    char* _rNameHolder;
```

Comment:
Why not just make this holder a std::string?   Then you could just assign instead of strdup in _getRname, and the cleanup in the dtor would be automatic?